python_scripts/bhyve_next_watering.py

```diff
@@ -24,36 +24,39 @@
 logger.info("updating next_watering for zone: ({}: {})".format(zone_name, zone))
 
 next_watering_entity = f"sensor.{zone_name}_next_watering".replace(" ", "_").replace("-", "_").lower()
-next_watering_attrs = {"friendly_name": f"{zone_name} next watering"}
+next_watering_attrs = {"friendly_name": f"{zone_name} next watering", "device_class": "timestamp"}
 
 rain_delay_finishing_entity = f"sensor.{device_name}_rain_delay_finishing".replace(
     " ", "_"
 ).replace("-", "_").lower()
-rain_delay_finishing_attrs = {"friendly_name": f"{device_name} rain delay finishing"}
+rain_delay_finishing_attrs = {"friendly_name": f"{device_name} rain delay finishing", "device_class": "timestamp"}
 
-rain_delay = hass.states.get(f"switch.{device_name}_rain_delay")
+rain_delay = hass.states.get(f"switch.{device_name}_rain_delay".replace(
+    " ", "_"
+).replace("-", "_").lower())
 
 if zone.state == "unavailable":
     hass.states.set(next_watering_entity, "Unavailable", next_watering_attrs)
     hass.states.set(
         rain_delay_finishing_entity, "Unavailable", rain_delay_finishing_attrs
     )
 else:
-    delay_finishes_at = None
-    next_watering = None
+    delay_finishes_at = 'unknown'
+    next_watering = 'unknown'
```

Comment:
It would be preferable to extract out a constant for the `unknown` string literal

Reply:
Looks like you can't, unfortunately... The only variables available are `hass`, `data`, and `logger`, and "it is not possible to use Python imports."

[[HA Python Scripts docs](https://www.home-assistant.io/integrations/python_script/)]

Reply:
Haha... I meant just a const variable at the top of the script 😃